app/javascript/components/home/HomePage.jsx

```diff
@@ -27,8 +33,16 @@ export default function HomePage() {
     [currentUser.signed_in],
   );
 
+  useEffect(() => {
+    document.cookie = `token=${inviteToken};path=/;`;
+  }, [inviteToken]);
+
   if (isLoading) return <Spinner />;
 
+  function showSignUp() {
+    return registrationMethod !== 'invite' || !!inviteToken;
```

Comment:
Kind of hard to follow.
I would personally prefer a `===` operator instead and adding an extra clause.
SignUp will show up if the registration method is "Open" or "Approval" OR if the inviteToken (the URL param) is blank?

Reply:
Yeah the double bang is slightly annoying. It won't show Signup if the registration method is invite and there is no invite token. I'll add a comment here

Reply:
Agreed,
Besides, in JS there's no need to cast a value to a boolean for conditional checks,
`!!inviteToken` should be the same as `inviteToken` in a condition,